fastly/block_fastly_service_v1_waf.go

```diff
@@ -160,21 +170,40 @@ func buildCreateWAF(WAFMap interface{}, serviceID string, ServiceVersion string)
 
 func buildDeleteWAF(WAFMap interface{}, ServiceVersion string) *gofastly.DeleteWAFInput {
 	df := WAFMap.(map[string]interface{})
+
+	version, _ := strconv.Atoi(ServiceVersion)
+
 	opts := gofastly.DeleteWAFInput{
-		ID:      df["waf_id"].(string),
-		Version: ServiceVersion,
+		ID:             df["waf_id"].(string),
+		ServiceVersion: version,
 	}
 	return &opts
 }
 
-func buildUpdateWAF(wafMap interface{}, serviceID string, ServiceVersion string) *gofastly.UpdateWAFInput {
+func buildUpdateWAF(d *schema.ResourceData, wafMap interface{}, serviceID string, ServiceVersion string) *gofastly.UpdateWAFInput {
 	df := wafMap.(map[string]interface{})
-	opts := gofastly.UpdateWAFInput{
-		Service:           serviceID,
-		Version:           ServiceVersion,
-		ID:                df["waf_id"].(string),
-		PrefetchCondition: df["prefetch_condition"].(string),
-		Response:          df["response_object"].(string),
+
+	version, _ := strconv.Atoi(ServiceVersion)
+
+	input := gofastly.UpdateWAFInput{
+		ServiceID:      gofastly.String(serviceID),
+		ServiceVersion: gofastly.Int(version),
+		ID:             df["waf_id"].(string),
 	}
-	return &opts
+
+	// NOTE: to access the WAF data we need to link to specific list index. This
+	// isn't ideal and if we already know that a WAF can only have MaxItems: 1,
+	// then we should probably move away from it being a schema.TypeList.
```

Comment:
Cool. Will add 👍🏻

Reply:
hmm, interesting thought: what would the schema be set to? 

I had presumed we'd move from `TypeList` to `TypeMap`, but according to the docs that would be `map[string]interface{}` and also, where the current TypeList has the `Elem` field set to a `schema.Resource` (whose own Schema field is `map[string]*schema.Schema`), the documentation suggests...

> For a TypeMap, Elem must be a *Schema with a Type that is one of the primitives: TypeString, TypeBool, TypeInt, or TypeFloat.

Which sounds like you can then only have one 'type' as the value for each key? Which is also a bit confusing because the `TypeMap` should be `map[string]interface{}` from what I can tell 🤔

Reply:
I might just reword the comment instead so it's clearer why we use the weird dot notation syntax, rather than state we should move away from the TypeList (because right now it might be 'by design')